src/components/SnapshotButton.tsx

```diff
@@ -24,22 +24,32 @@
 
 import { RefObject } from "react";
 import CameraAltIcon from "@mui/icons-material/CameraAlt";
+import { default as OlMap } from "ol/Map";
 
 import i18n from "@/i18n";
 import { WithLocale } from "@/util/lang";
 import { MessageType } from "@/states/messageLogState";
 import { exportElement } from "@/util/export";
 import ToolButton from "@/components/ToolButton";
+import { MAP_OBJECTS } from "@/states/controlState";
```

Comment:
Wrong dependency. Create a higher level component `MapSnapshotButton` instead that uses `SnapshotButton`.

Reply:
Make `MapSnapshotButton ` a private component of `MapInteractionsBar`.